storefront/app/controllers/spree/line_items_controller.rb

```diff
@@ -2,7 +2,10 @@ module Spree
   class LineItemsController < Spree::StoreController
     include CartMethods
 
-    before_action :require_user, unless: -> { current_store.prefers_guest_checkout? }
+    before_action unless: -> { current_store.prefers_guest_checkout? } do
```

Comment:
@mad-eel let's not block adding/creating carts, we just need to block checkout page access and redirect the visitor to signup page when they click "checkout" from cart